File: mock_data/Product.py
from Record import *
class Product:
	def __init__(self,name,category,price):
		self.name = name
		self.category = category
		self.price = price
		self.salesRecord = []
		self.shipmentsRecord = []

	def addSales(self,sales):
		self.salesRecord.append(sales)
	
	def addShipment(self,shipment):
		self.shipmentsRecord.append(shipment)

	def printDetails(self):
		print self.name + ", " + self.category + ", " + str(self.price)
		for sales in self.salesRecord:
			sales.printDetails()
		for shipment in self.shipmentsRecord:
			shipment.printDetails()

	def getNumOfSales(self):
		return len(self.salesRecord)

File: ethereum/src/main/java/com/crumbs/entities/Product.java
package com.crumbs.entities;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Set;

/**
 * Created by low on 2/2/17 11:41 PM.
 */
@Table (name = "products")
@Entity
@Getter
@Setter
public class Product implements Serializable {

	@Id
	private String name;
	private String category;
	private long price;
	@OneToMany (mappedBy = "product", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
	private Set<SalesRecord> salesRecord;
	@OneToMany (mappedBy = "product", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
	private Set<Shipment> shipmentsRecord;
}
